File: rig4j/src/main/java/com/alflabs/rig4/struct/BlogEntry.java
package com.alflabs.rig4.struct;

import com.alflabs.annotations.NonNull;
import com.alflabs.rig4.blog.BlogConfig;
import com.google.auto.value.AutoValue;

/**
 * A blog entry in the main index file: "blog config-number gdoc-file-id".
 * <p/>
 * This value structure indicates which gdoc file id to read and
 * to which {@link BlogConfig} it corresponds.
 */
@AutoValue
public abstract class BlogEntry {
    public static BlogEntry create(@NonNull String fileId, int configNumber) {
        return new AutoValue_BlogEntry(fileId, configNumber);
    }

    public abstract String getFileId();
    public abstract int getConfigNumber();
}

File: rig4k/src/main/kotlin/com/alflabs/rig4k/site/BlogEntry.kt
package com.alflabs.rig4k.site

import com.alflabs.rig4k.dl.BlogEntity

/**
 * A blog entry in the main index file: "blog config-number gdoc-file-id".
 *
 * This value structure indicates which gdoc file id to read and
 * to which [BlogConfig] it corresponds.
 */
data class BlogEntry(
    val entity: BlogEntity,
    val configNumber: Int,
)
